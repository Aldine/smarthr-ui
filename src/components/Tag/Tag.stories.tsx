import { storiesOf } from '@storybook/react'
import * as React from 'react'
import styled from 'styled-components'

import { Tag } from './Tag'
import readme from './README.md'

storiesOf('Tag', module)
  .addParameters({
    readme: {
      sidebar: readme,
    },
  })
  .add('all', () => (
    <List>
      <li>
        <Tag>デフォルト</Tag>
      </li>
      <li>
        <Tag type="success">サクセス</Tag>
      </li>
      <li>
        <Tag type="warning">警告</Tag>
      </li>
      <li>
        <Tag type="error">エラー</Tag>
      </li>
      <li>
        <Tag type="require">必須</Tag>
      </li>
      <li>
        <Tag type="success" skeleton={true}>
          サクセス
        </Tag>
      </li>
      <li>
        <Tag type="warning" skeleton={true}>
          警告
        </Tag>
      </li>
      <li>
        <Tag type="error" skeleton={true}>
          エラー
        </Tag>
      </li>
      <li>
        <Tag type="require" skeleton={true}>
          必須
        </Tag>
      </li>
    </List>
  ))

const List = styled.ul`
  padding: 8px 24px;
  list-style: none;
`
