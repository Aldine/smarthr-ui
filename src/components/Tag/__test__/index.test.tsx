import { mount } from 'enzyme'
import React from 'react'

import { Tag } from '../Tag'

describe('Tag', () => {
  it('should render given children', () => {
    const hello = 'hello'
    const wrapper = mount(<Tag type="success">{hello}</Tag>)

    expect(wrapper.text()).toBe(hello)
  })
})
