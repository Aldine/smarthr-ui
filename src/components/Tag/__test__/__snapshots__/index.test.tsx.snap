// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should be match snapshot 1`] = `
<Component
  type="success"
>
  <TagComponent
    theme={
      Object {
        "frame": Object {
          "border": Object {
            "default": "1px solid #d6d6d6",
            "lineStyle": "solid",
            "lineWidth": "1px",
            "radius": Object {
              "l": "8px",
              "m": "6px",
              "s": "3px",
            },
          },
        },
        "interaction": Object {
          "hover": Object {
            "animation": ".3s ease-out",
            "animationDuration": ".3s",
            "animationTiming": "ease-out",
            "feedbackOpacity": ".7",
          },
        },
        "palette": Object {
          "Background": "#f5f6fa",
          "Background_Omen": "#009ea6",
          "Background_Omen_M07": "#0dbac1",
          "Black": "#333",
          "Blue": "#0081cc",
          "Blue_M20": "#4dbdff",
          "Blue_P30": "#005180",
          "GreenGray": "#6bb0b3",
          "GreenGray_M20": "#f0f9fa",
          "Main": "#00a5ab",
          "Main_P10": "#008d91",
          "Main_P20": "#007378",
          "Mono_P03": "#f9f9f9",
          "Mono_P05": "#f5f5f5",
          "Mono_P10": "#eaeaea",
          "Mono_P20": "#d6d6d6",
          "Mono_P30": "#c1c1c1",
          "Mono_P40": "#adadad",
          "Mono_P60": "#858585",
          "Mono_P80": "#5c5c5c",
          "Orange": "#f80",
          "Orange_M30": "#ffac4d",
          "Orange_P10": "#e67a00",
          "Overlay": "rgba(51, 51, 51, 0.4)",
          "Red": "#ef475b",
          "SmartHRGreen": "#00c4cc",
          "White": "#fff",
          "Yellow": "#fc0",
        },
        "size": Object {
          "font": Object {
            "grande": 16,
            "short": 13,
            "tall": 14,
            "tasting": 12,
            "trenta": 22,
            "venti": 20,
          },
          "mediaQuery": Object {
            "sp": 599,
            "tablet": 959,
          },
          "pxToRem": [Function],
          "space": Object {
            "l": 40,
            "m": 32,
            "s": 24,
            "xl": 48,
            "xs": 16,
            "xxl": 56,
            "xxs": 8,
          },
        },
      }
    }
    type="success"
  >
    <styled.span
      className="success"
      skeleton={false}
      theme={
        Object {
          "frame": Object {
            "border": Object {
              "default": "1px solid #d6d6d6",
              "lineStyle": "solid",
              "lineWidth": "1px",
              "radius": Object {
                "l": "8px",
                "m": "6px",
                "s": "3px",
              },
            },
          },
          "interaction": Object {
            "hover": Object {
              "animation": ".3s ease-out",
              "animationDuration": ".3s",
              "animationTiming": "ease-out",
              "feedbackOpacity": ".7",
            },
          },
          "palette": Object {
            "Background": "#f5f6fa",
            "Background_Omen": "#009ea6",
            "Background_Omen_M07": "#0dbac1",
            "Black": "#333",
            "Blue": "#0081cc",
            "Blue_M20": "#4dbdff",
            "Blue_P30": "#005180",
            "GreenGray": "#6bb0b3",
            "GreenGray_M20": "#f0f9fa",
            "Main": "#00a5ab",
            "Main_P10": "#008d91",
            "Main_P20": "#007378",
            "Mono_P03": "#f9f9f9",
            "Mono_P05": "#f5f5f5",
            "Mono_P10": "#eaeaea",
            "Mono_P20": "#d6d6d6",
            "Mono_P30": "#c1c1c1",
            "Mono_P40": "#adadad",
            "Mono_P60": "#858585",
            "Mono_P80": "#5c5c5c",
            "Orange": "#f80",
            "Orange_M30": "#ffac4d",
            "Orange_P10": "#e67a00",
            "Overlay": "rgba(51, 51, 51, 0.4)",
            "Red": "#ef475b",
            "SmartHRGreen": "#00c4cc",
            "White": "#fff",
            "Yellow": "#fc0",
          },
          "size": Object {
            "font": Object {
              "grande": 16,
              "short": 13,
              "tall": 14,
              "tasting": 12,
              "trenta": 22,
              "venti": 20,
            },
            "mediaQuery": Object {
              "sp": 599,
              "tablet": 959,
            },
            "pxToRem": [Function],
            "space": Object {
              "l": 40,
              "m": 32,
              "s": 24,
              "xl": 48,
              "xs": 16,
              "xxl": 56,
              "xxs": 8,
            },
          },
        }
      }
    >
      <StyledComponent
        className="success"
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "ddLWBq",
              "rules": Array [
                "
  ",
                [Function],
                "
",
              ],
            },
            "displayName": "styled.span",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "span",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        skeleton={false}
        theme={
          Object {
            "frame": Object {
              "border": Object {
                "default": "1px solid #d6d6d6",
                "lineStyle": "solid",
                "lineWidth": "1px",
                "radius": Object {
                  "l": "8px",
                  "m": "6px",
                  "s": "3px",
                },
              },
            },
            "interaction": Object {
              "hover": Object {
                "animation": ".3s ease-out",
                "animationDuration": ".3s",
                "animationTiming": "ease-out",
                "feedbackOpacity": ".7",
              },
            },
            "palette": Object {
              "Background": "#f5f6fa",
              "Background_Omen": "#009ea6",
              "Background_Omen_M07": "#0dbac1",
              "Black": "#333",
              "Blue": "#0081cc",
              "Blue_M20": "#4dbdff",
              "Blue_P30": "#005180",
              "GreenGray": "#6bb0b3",
              "GreenGray_M20": "#f0f9fa",
              "Main": "#00a5ab",
              "Main_P10": "#008d91",
              "Main_P20": "#007378",
              "Mono_P03": "#f9f9f9",
              "Mono_P05": "#f5f5f5",
              "Mono_P10": "#eaeaea",
              "Mono_P20": "#d6d6d6",
              "Mono_P30": "#c1c1c1",
              "Mono_P40": "#adadad",
              "Mono_P60": "#858585",
              "Mono_P80": "#5c5c5c",
              "Orange": "#f80",
              "Orange_M30": "#ffac4d",
              "Orange_P10": "#e67a00",
              "Overlay": "rgba(51, 51, 51, 0.4)",
              "Red": "#ef475b",
              "SmartHRGreen": "#00c4cc",
              "White": "#fff",
              "Yellow": "#fc0",
            },
            "size": Object {
              "font": Object {
                "grande": 16,
                "short": 13,
                "tall": 14,
                "tasting": 12,
                "trenta": 22,
                "venti": 20,
              },
              "mediaQuery": Object {
                "sp": 599,
                "tablet": 959,
              },
              "pxToRem": [Function],
              "space": Object {
                "l": 40,
                "m": 32,
                "s": 24,
                "xl": 48,
                "xs": 16,
                "xxl": 56,
                "xxs": 8,
              },
            },
          }
        }
      >
        <span
          className="success sc-bdVaJa ddLWBq"
        >
          hello
        </span>
      </StyledComponent>
    </styled.span>
  </TagComponent>
</Component>
`;
